tests/jointcalTestBase.py

```diff
@@ -255,3 +258,122 @@ def _test_metrics(self, result, expect):
                     self.assertFloatsAlmostEqual(value, expect[key.metric], msg=key.metric, rtol=1e-5)
                 else:
                     self.assertEqual(value, expect[key.metric], msg=key.metric)
+
+    def _importRepository(self, instrument, exportPath, exportFile):
+        """Import a gen3 test repository into self.testDir
+
+        Parameters
+        ----------
+        instrument : `str`
+            Full string name for the instrument.
+        exportPath : `str`
+            Path to location of repository to export.
+            This path must contain an `exports.yaml` file containing the
+            description of the exported gen3 repo that will be imported.
+        exportFile : `str`
+            Filename of export data.
+        """
+        self.repo = os.path.join(self.output_dir, 'testrepo')
+
+        # Make the repo and retrieve a writeable Butler
+        _ = lsst.daf.butler.Butler.makeRepo(self.repo)
+        butler = lsst.daf.butler.Butler(self.repo, writeable=True)
+        # Register the instrument
+        instrInstance = lsst.obs.base.utils.getInstrument(instrument)
+        instrInstance.register(butler.registry)
+        # Import the exportFile
+        butler.import_(directory=exportPath, filename=exportFile,
+                       transfer='symlink',
+                       skip_dimensions={'instrument', 'detector', 'physical_filter'})
+
+    def _runPipeline(self, repo, queryString=None,
```

Comment:
I'm not sure why you don't want to use pipeline files.  It's how the system is supposed to be run (though I suppose in a test it can be simulated, that's swimming upstream).

Reply:
Ok, I'll add it back in on DM-27869, where I can actually try it out.